src/sql/workbench/contrib/notebook/browser/models/notebookModel.ts

```diff
@@ -997,4 +1031,160 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		this._contentChangedEmitter.fire(changeInfo);
 	}
 
+	findNext(): Promise<NotebookRange> {
```

Comment:
I would love to have these methods live somewhere else instead of putting it into the notebook model. Maybe into a notebookFindModel?

The notebook model should focus on a model for how to hold and change properties of the notebook itself, whereas when we use find, we're not inherently changing anything about the notebook itself. Does that make sense?

Reply:
I've created a separate interface for find and moved the implementation to notebookFindModel. So notebookModel has no find logic anymore.